src/sql/parts/connection/common/connectionStatusManager.ts

```diff
@@ -194,8 +194,8 @@ export class ConnectionStatusManager {
 	 * Get a list of the active connection profiles managed by the status manager
 	*/
 	public getActiveConnectionProfiles(): ConnectionProfile[] {
-		let profiles = Object.values(this._connections).map((connectionInfo: ConnectionManagementInfo) => connectionInfo.connectionProfile);
+		let profiles = Object.values(this._connections).filter((connectionInfo: ConnectionManagementInfo) => !connectionInfo.deleted).map((connectionInfo: ConnectionManagementInfo) => connectionInfo.connectionProfile);
```

Comment:
The TSDoc comment for `ConnectionManagementInfo.deleted` says "Whether the connection should be deleted after connection is complete."

As far as I know we use that for temporary connections that aren't intended to be saved in Object Explorer so I'm surprised this would address the issue you're seeing. Does everything look good in your testing?

Reply:
Yeah this fixed the issues I was dealing with.